primitives/src/services/mod.rs

```diff
@@ -769,14 +770,64 @@ impl PriceTargets {
 }
 
 #[derive(PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, Copy, Clone, MaxEncodedLen)]
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct OperatorPreferences {
 	/// The operator ECDSA public key.
-	pub key: ecdsa::Public,
+	pub key: [u8; 65],
 	/// The pricing targets for the operator's resources.
 	pub price_targets: PriceTargets,
 }
 
+#[cfg(feature = "std")]
+impl Serialize for OperatorPreferences {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		use serde::ser::SerializeTuple;
+		let mut tup = serializer.serialize_tuple(2)?;
+		tup.serialize_element(&self.key[..])?;
+		tup.serialize_element(&self.price_targets)?;
+		tup.end()
+	}
+}
+
+#[cfg(feature = "std")]
+struct OperatorPreferencesVisitor;
+
+#[cfg(feature = "std")]
+impl<'de> serde::de::Visitor<'de> for OperatorPreferencesVisitor {
+	type Value = OperatorPreferences;
+
+	fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result {
+		formatter.write_str("a tuple of 2 elements")
+	}
+
+	fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+	where
+		A: serde::de::SeqAccess<'de>,
+	{
+		let key = seq
+			.next_element::<Vec<u8>>()?
+			.ok_or_else(|| serde::de::Error::custom("key is missing"))?;
+		let price_targets = seq
+			.next_element::<PriceTargets>()?
+			.ok_or_else(|| serde::de::Error::custom("price_targets is missing"))?;
+		let mut key_arr = [0u8; 65];
+		key_arr.copy_from_slice(&key);
```

Comment:
This here would panic instead of returning an error, I would recommend using `try_into` then map the error.
```suggestion
		let key_arr: [u8; 65] = key.try_into()
		.map_err(|_| serde::de::Error::cusom("key must be in the uncompressed format with length of 65 bytes");
```